File: lib/features/finance/domain/entities/finance.dart
class Finance {
  final int id;
  final String type; // e.g., 'income', 'expense'
  final double amount;
  final String description;
  final String date;

  Finance({
    required this.id,
    required this.type,
    required this.amount,
    required this.description,
    required this.date,
  });
}

File: backend/app/Models/Finance.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Finance extends Model
{
    protected $fillable = [
        'type',
        'amount',
        'description',
        'date',
        'company_id',
    ];

    protected $casts = [
        'amount' => 'decimal:2',
        'date' => 'date',
    ];

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }
}
